### PR TITLE
vma: Add !VVAR condition to vma_entry_can_be_lazy

### PR DESCRIPTION
**Pull request by yota9**
Currently most of the times we don't have problems with VVAR segment and
lazy restore because when VDSO is parked there is an munmap call that
calls UFFDIO_UNREGISTER on the destination address.
But we don't want to enable userfaultfd for VDSO and VVAR at the first
place.

### CODE REVIEW

**Comment by avagin** [comment]
lgtm

**Comment by rppt** [comment]
>  AFAIU Currently most of the times we don't have problems with VVAR segment due
> to during parking vdso on it's old place mremap is used and during it
> munmap is called to destination address which results in calling
> UFFDIO_UNREGISTER. But as for VDSO we don't want to enable uffd for these addresses at firse place.

I find the changelog a bit hard to parse, how about

Currently most of the times we don't have problems with VVAR segment and lazy restore because when VDSO is parked there is an munmap call that calls UFFDIO_UNREGISTER on the destination address.
But we don't want to enable userfaultfd for VDSO and VVAR at the first place.

Other than that, LGTM

**Comment by yota9** [comment]
@rppt Thanks, done. Please merge it since I don't have rights.

**Review by rppt — APPROVED** [review]
